lib/net.js

```diff
@@ -899,24 +899,18 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
-Socket.prototype.connect = function(options, cb) {
+Socket.prototype.connect = function() {
+  const args = new Array(arguments.length);
+  for (var i = 0; i < arguments.length; i++)
+    args[i] = arguments[i];
+  // TODO(joyeecheung): use destructuring when V8 is fast enough
```

Comment:
this code is same code as original line number 62 to 68.
and while doing this, i also considered `exports.createConnection` this function to just create `new Socket` and call `connect` by just forwarding arguments (for higher code coverage, if it is okay to move `options.timeout` also within `Socket.prototype.connect`.

e.g.

```javascript
...

exports.connect = exports.createConnection = function(options, cb) {
  const socket = new Socket(options);
  return Socket.prototype.connect.apply(socket, options, cb);
};

...

Socket.prototype.connect = function() {
  const args = new Array(arguments.length);
  for (var i = 0; i < arguments.length; i++)
    args[i] = arguments[i];
  // TODO(joyeecheung): use destructuring when V8 is fast enough
  const normalized = normalizeArgs(args);
  const options = normalized[0];
  const cb = normalized[1];
  debug('Socket.connect', normalized);

  if (options.timeout)
    this.setTimeout(options.timeout);

  if (this.write !== Socket.prototype.write)
    this.write = Socket.prototype.write;

...
```

what do you think @joyeecheung